### ISSUE TITLE
FullPath in OriginalFileName and PublishedName shouldn't be used when comparing DismDriverPackage

### ISSUE THREAD

**Issue by elf2k00** [jeffkl/ManagedDism#221]
The Equals override implementation for DismDriverPackage uses all the properties to compare a DismDriverPackage object with another, but even when they are equal they might have been already installed (and these properties changed) or its files be located in another folder.

``` 
public bool Equals(DismDriverPackage? other)
        {
            return other != null
                   && BootCritical == other.BootCritical
                   && InBox == other.InBox
                   && CatalogFile == other.CatalogFile
                   && ClassDescription == other.ClassDescription
                   && ClassGuid == other.ClassGuid
                   && ClassName == other.ClassName
                   && Date == other.Date
                   && DriverSignature == other.DriverSignature
                   && OriginalFileName == other.OriginalFileName
                   && ProviderName == other.ProviderName
                   && PublishedName == other.PublishedName;
        }
```

For example a comparison for the same driver ```vm3d.inf```, one read from an offline DriverStore and the other read from the downloaded driver returns false when they are the same driver.

DriverStore:
> Original Filename: G:\Windows\System32\DriverStore\FileRepository\vm3d.inf_amd64_9d22a6e67525b799\vm3d.inf
> Published Name:    oem1.inf

Downloaded Driver: 
> Original Filename: C:\drivers\vmware3d\vm3d.inf
> Published Name:    vm3d.inf

Filename and not the Full Path should be used for comparing ```OriginalFilename```, and ```PublishedName``` shouldn't be used at all because the .inf name changes when installed and might be different between installations.

**Comment by hdensity** [comment]
Not trying to rehash the age-old _domain-aware_ against _domain-agnostic_ equals and hash-code debate, and I do not speak for @jeffkl, but my personal preference is the latter. The responsibility of Equals and GetHashCode should remain limited to ensuring ValueType equality between class instances. As an aside, I would go as far as changing the classes to _records_ and rely on the compiler synthesized ValueType Equals and GetHashCode methods.

I do understand your problem though, and you could solve it using a dedicated Comparer, or an _IsInstalled_ method, maybe as an extension for DismDriver(Package). Please be aware though, that DismGetDriverInfo may return different lists of DismDriver structures for the same DismDriverPackage, based on whether you call the method for a published (installed) driver or a local .inf file. See [DismDriver](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism/dismdriver-structure) for more information.

**Comment by jeffkl** [comment]
I tend to agree, `Equals()` is just seeing if two objects are equal and in this case its correct.  I'm interested to know about the scenario where you're comparing to `DismDriverPackage` instances?  If needed, I'm all for adding a new method that compares these objects in a more specific way or adding a implementation of `IComparer` and adding an overload to `Equals()`
